### PR TITLE
Updated backup cuts

### PR DESCRIPTION
**Pull request by apcooper**
This PR updates the backup target selection cuts to match the backup program document:
https://www.overleaf.com/project/61116134e46f88886a3367d2

- Add bp-rp based mag thresholds
- Add new backup_giant target class

### CODE REVIEW

**Comment by apcooper** [comment]
@geordie666 I think this is ready for you to take a look, thanks.

**Review by geordie666 — CHANGES_REQUESTED** [review]
This all looks good @apcooper, thanks. Thanks, in particular, for taking care to be backwards-compatible with `sv3`. I've run a handful of example files using your branch and put them in:
```
/global/cscratch1/sd/adamyers/gaiadr2/1.3.0.dev5218/targets/main/resolve/backup
/global/cscratch1/sd/adamyers/gaiadr2/1.3.0.dev5218/targets/sv3/resolve/backup
```
Everything ran to completion, exactly as I would have expected. I ran with the `--gaiasub` option for `main`, meaning that those results should use Gaia edr3.

I've requested a few minor changes for style and documentation (and also included a question I had about behavior near the Galactic Plane). Once those changes are addressed, I'm certainly happy for this to be merged. Once we merge this, I can run a larger set of example files for people to vet.

File: py/desitarget/cuts.py [hunk 1]
```diff
@@ -360,6 +365,9 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
     :class:`array_like`
         ``True`` if and only if the object is a very faint "BACKUP"
         target.
+    :class:`array_like`
+        ``True`` if and only if the object is a gaia-selected giant
+        "BACKUP" target.
 
     Notes
     -----
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:373]: Done.

File: py/desitarget/cuts.py [hunk 2]
```diff
@@ -368,33 +376,55 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
...
+    # APC In this case the BP-RP relations use the fluxes without
+    # APC any correction for extinction, by design.
+    bprp = gaiabmag - gaiarmag
+
     # ADM restrict all classes to dec >= -30.
     primary &= dec >= -30.
+    # APC require measured  gaia color.
+    primary &= ~np.isnan(bprp)
+    # APC hard bright limit
+    primary &= gaiagmag > 10.0
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:388]: Done.

File: py/desitarget/cuts.py [hunk 3]
```diff
@@ -368,33 +376,55 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
...
 
     # ADM determine which sources are close to the Galaxy.
     in_gal = is_in_Galaxy([ra, dec], radec=True)
 
-    # ADM bright targets are 10 < G < 16.
-    isbackupbright &= gaiagmag >= 10
-    isbackupbright &= gaiagmag < 16
+    # APC bright targets are 11.2 + 0.6(BP-RP) < G < 16.
+    isbackupbright &= gaiagmag > 11.2 + 0.6*bprp
+    isbackupbright &= gaiagmag <= 16.0
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:400]: Done.

File: py/desitarget/cuts.py [hunk 4]
```diff
@@ -368,33 +376,55 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
...
+
+    # APC Giant candidates have low parallax
+    is_gaiagiant = parallax < (3 * parallaxerr + 0.1)
 
-    # ADM faint targets are 16 < G < 18.
-    isbackupfaint &= gaiagmag >= 16
-    isbackupfaint &= gaiagmag < 18.
+    # APC giant targets are min(17.5 + 0.6(BP-RP), 19) < G < 16
+    isbackupgiant &= gaiagmag > 16.0
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:406]: Done.

File: py/desitarget/cuts.py [hunk 5]
```diff
@@ -368,33 +376,55 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
...
+    # APC and are likely giants
+    isbackupgiant &= is_gaiagiant
+
+    # APC faint targets are 16 < G < 18
+    isbackupfaint &= gaiagmag > 16.0
+    isbackupfaint &= gaiagmag <= 18.0
+    # APC and are not halo giant candidates
+    isbackupfaint &= ~is_gaiagiant
     # ADM and are "far from" the Galaxy.
     isbackupfaint &= ~in_gal
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:417]: As you say the original point of those geometric cuts was to keep the target catalog small. The discussion in https://github.com/desihub/desitarget/issues/541 suggests that might not be an issue now? If not, we can probably remove this `~in_gal` cut.

File: py/desitarget/cuts.py [hunk 6]
```diff
@@ -368,33 +376,55 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
...
 
     # ADM very faint targets are 18. < G < 19.
     isbackupveryfaint &= gaiagmag >= 18.
     isbackupveryfaint &= gaiagmag < 19
+    # APC and are not halo giant candidates
+    isbackupveryfaint &= ~is_gaiagiant
     # ADM and are "far from" the Galaxy.
     isbackupveryfaint &= ~in_gal
```
[the remark this answers is not in the document]
Comment (apcooper) [reply py/desitarget/cuts.py:425]: See comment above.

**Comment by geordie666** [comment]
I'm going to leave this open a little longer; @schlafly wants to check some bright-end cuts in G, Bp, Rp, in case of saturation. So we may need to add a few more lines of code before running this.

**Comment by geordie666** [comment]
@apcooper : Do Eddie's suggested bright-end cuts look good? If so, I'll add those cuts, merge this PR, and then run an example set of targets. Cori is down tomorrow, so it might be Thursday before we have the files.

**Comment by apcooper** [comment]
Thanks. Looks OK but I'd like @segasai's opinion.

**Comment by geordie666** [comment]
Thanks, all. Unless I hear otherwise, once Cori is back up (or more likely tomorrow) I'll add Eddie's proposed bright limits and merge this PR. 

@apcooper: As Cori will be back up sooner, your-time (I _think_), feel free to beat me to this and add Eddie's limits yourself!

**Pull request merged**